### ISSUE TITLE
WebSurfer resources are not destroyed when runtime stops

### ISSUE THREAD

**Issue by husseinmozannar** [microsoft/autogen#4699]
The websurfer browser is not being destroyed when the runtime exits. This becomes apparent in AutoGenStudio and when you have a loop where you create a web agent. The browser is only destroyed when the whole program is interrupted but not a process.

Code to replicate:

```python
import asyncio
from autogen_agentchat.ui import Console
from autogen_agentchat.teams import RoundRobinGroupChat
from autogen_ext.models import OpenAIChatCompletionClient
from autogen_ext.agents.web_surfer import MultimodalWebSurfer




async def main_websurfer() -> None:
    # Define an agent
    web_surfer_agent = MultimodalWebSurfer(
        name="MultimodalWebSurfer",
        model_client=OpenAIChatCompletionClient(model="gpt-4o-2024-08-06"),
        headless=False,
    )

    # Define a team
    agent_team = RoundRobinGroupChat([web_surfer_agent], max_turns=1)

    # Run the team and stream messages to the console
    stream = agent_team.run_stream(task="Navigate to the AutoGen readme on GitHub.")
    await Console(stream)

if __name__ == "__main__":
    for _ in range(3):
        asyncio.run(main_websurfer())
```

How many browsers do you expect to pop up: 1 
how many actually pop up: 3

**Comment by ekzhu** [comment]
@husseinmozannar this code:

```python
# Define an agent
    web_surfer_agent = MultimodalWebSurfer(
        name="MultimodalWebSurfer",
        model_client=OpenAIChatCompletionClient(model="gpt-4o-2024-08-06"),
        headless=False,
    )
```

Creates a reference to the web surfer agent, this reference is to an object that the runtime also uses. Unlike an agent in the Core API, the runtime does not take over this reference and its lifecycle, it stays with the application program it is running inside. 

What we need to do here is to properly release and close the resource used by this agent upon existing the `main_websurfer` function in your code snippet. In general, we should always close this resource when application exits. 

When the resource takes too long to initialize, consider the followings:
1. can we serialize and deserialize an initialized state of the resource, so we don't need to perform full initialization on every single request?
2. can we create a resource pool at the application level to manage these resources, and allocate to an agent upon request, and release from an agent once it is done with it. Always clean up the resource once the application exits.

**Comment by ekzhu** [comment]
Step 1 is to add a `close` method to web surfer. 

Let's focus on this for 0.4.

**Comment by husseinmozannar** [comment]
Temporary fix with close method, future work to be done on runtime to shutdown ressources

**Comment by victordibia** [comment]
@ekzhu ,

> let's remember to call the close method for web surfer by creating an asyncio task that triggers on an asyncio.event that fires upon closing the request handler, 


This is how its done currently in AGS. 
Each run is facilitated by a TeamManager, which has been updated as follows

```
team = await self._create_team(team_config, input_func)
result = await team.run(task=task, cancellation_token=cancellation_token)

# close agent resources
for agent in team._participants:
    if hasattr(agent, "close"):
        await agent.close()
```
